### ISSUE TITLE
ERROR! Invalid callback for stdout specified: awx_display

### ISSUE THREAD

**Issue by anxstj** [ansible/awx#12023]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.

### Summary

I tried to upgrade my AWX setup from 19.2.0 to 20.1.0. After the upgrade all jobs in the controlplane exit with the error `ERROR! Invalid callback for stdout specified: awx_display`. But it seems that jobs in the default queue (e.g. inventory updates) are still working.

I have a staging setup that uses the same installation method and the same image, but that is not affected. Only my production environment is affected. The main difference is that the production environment is upgraded from one version to the other and the staging environment is created from scratch if the AWX image changes.

### AWX version

20.1.0

### Select the relevant components

- [ ] UI
- [ ] API
- [ ] Docs

### Installation method

docker on linux

### Modifications

yes

### Ansible version

2.11.7

### Operating system

Debian 11

### Web browser

Firefox

### Steps to reproduce

Hard to tell. The only relevant change, except the container image, was the location of the receptor socket:

docker-compose.yml:
```
       - "./receptor.conf:/etc/receptor/receptor.conf"
       - "/sys/fs/cgroup:/sys/fs/cgroup"
       - "redis_socket_1:/var/run/redis/:rw"
-      - "receptor_1:/var/run/receptor/"
       - "./containers.conf:/etc/containers/containers.conf"
       - "/usr/local/share/ca-certificates:/etc/pki/ca-trust/source/anchors:ro"
     dns_search: "anx.local"
@@ -45,7 +44,5 @@
 volumes:
   awx_db:
     name: tools_awx_db
-  receptor_1:
-    name: tools_receptor_1
   redis_socket_1:
     name: tools_redis_socket_1
```
receptor.conf:
```
 
 - control-service:
     service: control
-    filename: /var/run/receptor/receptor.sock
+    filename: /var/run/awx-receptor/receptor.sock
 
 - local-only:
 ```

The AWX 20.1.0 docker image in staging and production is the same and the same is true for the awx-ee:latest image which is pulled by the control plane execution environment. So the only other changes are the receptor socket path and the persistent database.

### Expected results

The job executes correctly.

### Actual results

```
ERROR! Invalid callback for stdout specified: awx_display
```

### Additional information

Edit: The following seems not to be related, reverting the system back to 19.2.0 shows the same errors.
~~It seems that changing the receptor socket path has caused the following errors which flood the container log.~~
```
tools_awx_1 | 2022-04-08 07:23:37,956 ERROR    [-] awx.main.consumers Request user is not authenticated to use websocket.
tools_awx_1 | 2022-04-08 07:23:37,956 ERROR    [-] awx.main.consumers Request user is not authenticated to use websocket.
tools_awx_1 | 2022-04-08 07:23:37,956 ERROR    Request user is not authenticated to use websocket.
tools_awx_1 | 2022/04/08 07:23:38 [error] 71#71: *1118 recv() failed (104: Connection reset by peer) while proxying upgraded connection, client: 172.25.0.1, server: _, request: "GET /websocket/ HTTP/1.0", upstream: "http://127.0.0.1:8051/websocket/", host: "awx.example.com"
tools_awx_1 | 172.25.0.1 - - [08/Apr/2022:07:23:38 +0000] "GET /websocket/ HTTP/1.0" 101 21 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.84 Safari/537.36" "10.0.253.171
```

~~Changing the receptor socket back to the old location fixes that error, but the problem with the failing jobs persists.~~

Another observation was the receptor logs a warning when a job in the control plane fails:
```
DEBUG 2022/04/07 17:05:50 Client disconnected from control service @
WARNING 2022/04/07 17:05:50 Could not close connection: close unix /var/run/awx-receptor/receptor.sock->@: use of closed network connection
```

**Comment by shanemcd** [comment]
I think you need a newer version of ansible-runner. Try going in and flipping your EE to "always" pull to see if it fixes your problem. And if you've built any custom EEs, make sure you are building on top of the latest image.

**Comment by anxstj** [comment]
The controlplane queue was created with the start of the 20.1.0 AWX container as was the control plane execution environment which had quay.io/ansible/awx-ee:latest as image. And it pulled an image from yesterday (id 4ad3463f41ee). The same as the staging environment used.

**Comment by shanemcd** [comment]
Did you try to set the EE to always? Just because you are using the latest tag does not mean it will always pull from the registry. The "latest" image on-disk may be old. This is not pulled internally by AWX, the image lives on the underlying nod.

**Comment by shanemcd** [comment]
Maybe also try `image_pull_policy: Always` at the CR level https://github.com/ansible/awx-operator#deploying-a-specific-version-of-awx

**Comment by anxstj** [comment]
I changed the EE control plane settings to "Always pull". No change, a project update still fails with: "Invalid callback".

Inside the AWX container the following images are present:
```
$ podman images
REPOSITORY              TAG         IMAGE ID      CREATED       SIZE
quay.io/ansible/awx-ee  latest      f7816bdcd6e6  10 hours ago  1.7 GB
```
Version of ansible-runner inside the awx-ee container:
```
$ podman run -it quay.io/ansible/awx-ee:latest ansible-runner --version
2.1.3
```
ansible-runner version in AWX container:
```
$ /var/lib/awx/venv/awx/bin/ansible-runner --version
2.1.1.dev40
```

But on the other side, my staging environment has the same versions and is working. So that's not was caused the error.

**Comment by anxstj** [comment]
One more observation:

```
$ find /var/lib/awx/venv/ -name awx_display.py
/var/lib/awx/venv/awx/lib/python3.9/site-packages/ansible_runner/display_callback/callback/awx_display.py

$ podman run -it quay.io/ansible/awx-ee:latest  find / -name awx_display.py
/usr/local/lib/python3.8/site-packages/ansible_runner/callbacks/awx_display.py
```
The installation paths are different. But I don't know if this really matters.

Any advice on how I can debug this?

**Comment by anxstj** [comment]
I tried to trace the work a little bit more.

```
# receptorctl --socket /var/run/awx-receptor/receptor.sock work list
{'8JzGmTdq': {'Detail': 'Running: PID 470',
              'ExtraData': {'Params': 'worker '
                                      '--private-data-dir=/tmp/awx_787038_8xx_un1b',
                            'Pid': 462},
              'State': 1,
              'StateName': 'Running',
              'StdoutSize': 2229,
              'WorkType': 'local'}}
# receptorctl --socket /var/run/awx-receptor/receptor.sock work results 8JzGmTdq
{
    "status": "starting",
    "runner_ident": "787038",
    "command": [
        "podman",
        "run",
        "--rm",
        "--tty",
        "--interactive",
        "--workdir",
        "/runner/project",
        "-v",
        "/tmp/awx_787038_8xx_un1b/:/runner/:Z",
        "-v",
        "/tmp/ansible_runner_plugins_cnbuwwba/callback/:/home/runner/.ansible/plugins/callback/:Z",
        "-v",
        "/var/lib/awx/projects/_47__anxstj_awx_env_test/:/var/lib/awx/projects/_47__anxstj_awx_env_test/:Z",
        "-v",
        "/var/lib/awx/projects/.__awx_cache/_47__anxstj_awx_env_test/:/var/lib/awx/projects/.__awx_cache/_47__anxstj_awx_env_test/:Z",
        "--env-file",
        "/tmp/awx_787038_8xx_un1b/artifacts/787038/env.list",
        "--quiet",
        "--name",
        "ansible_runner_787038",
        "--user=root",
        "--network",
        "slirp4netns:enable_ipv6=true",
        "--pull=always",
        "quay.io/ansible/awx-ee:latest",
        "ssh-agent",
        "sh",
        "-c",
        "trap 'rm -f /runner/artifacts/787038/ssh_key_data' EXIT && ssh-add /runner/artifacts/787038/ssh_key_data && rm -f /runner/artifacts/787038/ssh_key_data && ansible-playbook -t update_git,install_roles,install_collections,delete -i /runner/inventory/hosts -e @/runner/env/extravars project_update.yml"
    ],
    "env": {
        "ANSIBLE_UNSAFE_WRITES": "1",
        "AWX_ISOLATED_DATA_DIR": "/runner/artifacts/787038",
        "ANSIBLE_FACT_CACHE_TIMEOUT": "0",
        "ANSIBLE_FORCE_COLOR": "True",
        "ANSIBLE_HOST_KEY_CHECKING": "False",
        "ANSIBLE_INVENTORY_UNPARSED_FAILED": "True",
        "ANSIBLE_PARAMIKO_RECORD_HOST_KEYS": "False",
        "HOME": "/var/lib/awx",
        "AWX_PRIVATE_DATA_DIR": "/tmp/awx_787038_8xx_un1b",
        "ANSIBLE_RETRY_FILES_ENABLED": "False",
        "ANSIBLE_ASK_PASS": "False",
        "ANSIBLE_BECOME_ASK_PASS": "False",
        "DISPLAY": "",
        "TMP": "/tmp",
        "PROJECT_UPDATE_ID": "787038",
        "ANSIBLE_GALAXY_SERVER_SERVER0_URL": "https://galaxy.ansible.com/",
        "ANSIBLE_GALAXY_SERVER_LIST": "server0",
        "ANSIBLE_STDOUT_CALLBACK": "awx_display",
        "RUNNER_OMIT_EVENTS": "False",
        "RUNNER_ONLY_FAILED_EVENTS": "False"
    },
    "cwd": "/runner/project"
}
{"status": "running", "runner_ident": "787038"}
{"event": "verbose", "uuid": "7545bb08-bda3-48b9-8bc4-c5d4370a7a17", "counter": 1, "stdout": "\u001b[33mWARN\u001b[0m[0000] \"/\" is not a shared mount, this could cause issues or missing mounts with rootless containers ", "start_line": 0, "end_line": 1, "runner_ident": "787038", "created": "2022-04-12T13:18:57.365048"}
{"event": "verbose", "uuid": "286d73f2-3d07-4588-aaa0-7b7d8706d4c7", "counter": 2, "stdout": "Enter passphrase for /runner/artifacts/787038/ssh_key_data: ", "start_line": 1, "end_line": 2, "runner_ident": "787038", "created": "2022-04-12T13:19:15.599272"}
{"event": "verbose", "uuid": "7fce3470-b911-40b6-88b2-0c09c238017a", "counter": 3, "stdout": "Identity added: /runner/artifacts/787038/ssh_key_data (ssh key)", "start_line": 2, "end_line": 3, "runner_ident": "787038", "created": "2022-04-12T13:19:15.748475"}
{"event": "verbose", "uuid": "a7b4b4e1-3461-451b-838e-8b2a9c5bbe15", "counter": 4, "stdout": "\u001b[0;31mERROR! Invalid callback for stdout specified: awx_display\u001b[0m", "start_line": 3, "end_line": 4, "runner_ident": "787038", "created": "2022-04-12T13:19:16.725293"}
{"status": "failed", "runner_ident": "787038"}
{"zipfile": 2092}
...
# ls -l /tmp/ansible_runner_plugins_cnbuwwba/callback/
total 36
-rw-r--r-- 1 awx root 30853 Apr 12 11:21 awx_display.py
-rw-r--r-- 1 awx root     0 Apr 12 11:21 __init__.py
drwxr-xr-x 2 awx root  4096 Apr 12 11:21 __pycache__
# diff /tmp/ansible_runner_plugins_cnbuwwba/callback/awx_display.py \
   /var/lib/awx/venv/awx/lib/python3.9/site-packages/ansible_runner/display_callback/callback/awx_display.py
# echo $?
0
```

**Comment by anxstj** [comment]
When I do the same on my staging node, I get the following results:

```
# receptorctl --socket /var/run/awx-receptor/receptor.sock work results pL1Hga1d
{
    "status": "starting",
    "runner_ident": "23",
    "command": [
        "podman",
        "run",
        "--rm",
        "--tty",
        "--interactive",
        "--workdir",
        "/runner/project",
        "-v",
        "/tmp/awx_23_nzel9o5k/:/runner/:Z",
        "-v",
        "/var/lib/awx/projects/_8__soplaysanx_baseconfig/:/var/lib/awx/projects/_8__soplaysanx_baseconfig/:Z",
        "-v",
        "/var/lib/awx/projects/.__awx_cache/_8__soplaysanx_baseconfig/:/var/lib/awx/projects/.__awx_cache/_8__soplaysanx_baseconfig/:Z",
        "--env-file",
        "/tmp/awx_23_nzel9o5k/artifacts/23/env.list",
        "--quiet",
        "--name",
        "ansible_runner_23",
        "--user=root",
        "--network",
        "slirp4netns:enable_ipv6=true",
        "--pull=always",
        "quay.io/ansible/awx-ee:latest",
        "ssh-agent",
        "sh",
        "-c",
        "trap 'rm -f /runner/artifacts/23/ssh_key_data' EXIT && ssh-add /runner/artifacts/23/ssh_key_data && rm -f /runner/artifacts/23/ssh_key_data && ansible-playbook -t update_git,install_roles,install_collections -i /runner/inventory/hosts -e @/runner/env/extravars project_update.yml"
    ],
    "env": {
        "LAUNCHED_BY_RUNNER": "1",
        "ANSIBLE_UNSAFE_WRITES": "1",
        "AWX_ISOLATED_DATA_DIR": "/runner/artifacts/23",
        "ANSIBLE_FORCE_COLOR": "True",
        "ANSIBLE_HOST_KEY_CHECKING": "False",
        "ANSIBLE_INVENTORY_UNPARSED_FAILED": "True",
        "ANSIBLE_PARAMIKO_RECORD_HOST_KEYS": "False",
        "AWX_PRIVATE_DATA_DIR": "/tmp/awx_23_nzel9o5k",
        "ANSIBLE_RETRY_FILES_ENABLED": "False",
        "ANSIBLE_ASK_PASS": "False",
        "ANSIBLE_BECOME_ASK_PASS": "False",
        "DISPLAY": "",
        "TMP": "/tmp",
        "PROJECT_UPDATE_ID": "23",
        "ANSIBLE_GALAXY_SERVER_SERVER0_URL": "https://galaxy.ansible.com/",
        "ANSIBLE_GALAXY_SERVER_LIST": "server0",
        "ANSIBLE_STDOUT_CALLBACK": "awx_display",
        "RUNNER_OMIT_EVENTS": "False",
        "RUNNER_ONLY_FAILED_EVENTS": "False"
    },
    "cwd": "/runner/project"
}
```
Comparing the two results shows that the staging node misses the callback mount:
```
/tmp/ansible_runner_plugins_cnbuwwba/callback/:/home/runner/.ansible/plugins/callback/:Z",
```
So, what causes the ansible-runner to mount the callback directory into the awx-ee container?

**Comment by anxstj** [comment]
Comparing the `env.list` files from the `artifacts` directory reveals, that the job on the staging node was started with `LAUNCHED_BY_RUNNER=1`. On the production node the `LAUNCHED_BY_RUNNER` variable is not defined (instead it has `HOME=/var/lib/awx` ?!?).

Then `/var/lib/awx/venv/awx/share/ansible-runner/utils/entrypoint.sh` shows that `ANSIBLE_CALLBACK_PLUGINS` is only exported if `LAUNCHED_BY_RUNNER` is defined.

`LAUNCHED_BY_RUNNER` will be set by ansible-runner itself if it detects that it runs in a containerized environment.

If I get it correctly then the work is passed from receptor to ansible-runner worker which starts the awx-ee with help of podman...  And the ansible-runner worker has "some magic" which adds the mount of the callback directory to podman.

**Comment by anxstj** [comment]
Changig
```
 "AWX_TASK_ENV": { "HOME": "/var/lib/awx" }
```
to 

```
 "AWX_TASK_ENV": {}
```
fixes the problem.

The job has still the additional callback dir mounted:
```
# receptorctl --socket /var/run/awx-receptor/receptor.sock work results 7F8qeMnq
{
    "status": "starting",
    "runner_ident": "787056",
    "command": [
        "podman",
        "run",
        "--rm",
        "--tty",
        "--interactive",
        "--workdir",
        "/runner/project",
        "-v",
        "/tmp/awx_787056_x_cyabrm/:/runner/:Z",
        "-v",
        "/tmp/ansible_runner_plugins_jb26w79j/callback/:/home/runner/.ansible/plugins/callback/:Z",
        "-v",
        "/var/lib/awx/projects/_47__anxstj_awx_env_test/:/var/lib/awx/projects/_47__anxstj_awx_env_test/:Z",
        "-v",
        "/var/lib/awx/projects/.__awx_cache/_47__anxstj_awx_env_test/:/var/lib/awx/projects/.__awx_cache/_47__anxstj_awx_env_test/:Z",
        "--env-file",
        "/tmp/awx_787056_x_cyabrm/artifacts/787056/env.list",
        "--quiet",
        "--name",
        "ansible_runner_787056",
        "--user=root",
        "--network",
        "slirp4netns:enable_ipv6=true",
        "--pull=always",
        "quay.io/ansible/awx-ee:latest",
        "ssh-agent",
        "sh",
        "-c",
        "trap 'rm -f /runner/artifacts/787056/ssh_key_data' EXIT && ssh-add /runner/artifacts/787056/ssh_key_data && rm -f /runner/artifacts/787056/ssh_key_data && ansible-playbook -t update_git,install_roles,install_collections,delete -i /runner/inventory/hosts -e @/runner/env/extravars project_update.yml"
    ],
    "env": {
        "ANSIBLE_UNSAFE_WRITES": "1",
        "AWX_ISOLATED_DATA_DIR": "/runner/artifacts/787056",
        "ANSIBLE_FACT_CACHE_TIMEOUT": "0",
        "ANSIBLE_FORCE_COLOR": "True",
        "ANSIBLE_HOST_KEY_CHECKING": "False",
        "ANSIBLE_INVENTORY_UNPARSED_FAILED": "True",
        "ANSIBLE_PARAMIKO_RECORD_HOST_KEYS": "False",
        "AWX_PRIVATE_DATA_DIR": "/tmp/awx_787056_x_cyabrm",
        "ANSIBLE_RETRY_FILES_ENABLED": "False",
        "ANSIBLE_ASK_PASS": "False",
        "ANSIBLE_BECOME_ASK_PASS": "False",
        "DISPLAY": "",
        "TMP": "/tmp",
        "PROJECT_UPDATE_ID": "787056",
        "ANSIBLE_GALAXY_SERVER_SERVER0_URL": "https://galaxy.ansible.com/",
        "ANSIBLE_GALAXY_SERVER_LIST": "server0",
        "ANSIBLE_STDOUT_CALLBACK": "awx_display",
        "RUNNER_OMIT_EVENTS": "False",
        "RUNNER_ONLY_FAILED_EVENTS": "False"
    },
    "cwd": "/runner/project"
}
...
{"status": "successful", "runner_ident": "787056"}
```

**Comment by anxstj** [comment]
It looks like this setting is very old:
```
$ psql awx -c 'select * from conf_setting where id = 1;'
 id |            created            |           modified           |     key      |          value           | user_id 
----+-------------------------------+------------------------------+--------------+--------------------------+---------
  1 | 2018-10-17 12:15:22.190743+00 | 2018-10-17 12:15:22.19076+00 | AWX_TASK_ENV | {"HOME": "/var/lib/awx"} |        
```

**Comment by anxstj** [comment]
To validate my finding, I set `"AWX_TASK_ENV": { "HOME": "/var/lib/awx" }` on the the staging system. And then triggered a project update:
```
{
    "status": "starting",
    "runner_ident": "29",
    "command": [
        "podman",
        "run",
        "--rm",
        "--tty",
        "--interactive",
        "--workdir",
        "/runner/project",
        "-v",
        "/tmp/awx_29_ywqhq5t9/:/runner/:Z",
        "-v",
        "/var/lib/awx/projects/_8__soplaysanx_baseconfig/:/var/lib/awx/projects/_8__soplaysanx_baseconfig/:Z",
        "-v",
        "/var/lib/awx/projects/.__awx_cache/_8__soplaysanx_baseconfig/:/var/lib/awx/projects/.__awx_cache/_8__soplaysanx_baseconfig/:Z",
        "--env-file",
        "/tmp/awx_29_ywqhq5t9/artifacts/29/env.list",
        "--quiet",
        "--name",
        "ansible_runner_29",
        "--user=root",
        "--network",
        "slirp4netns:enable_ipv6=true",
        "--pull=always",
        "quay.io/ansible/awx-ee:latest",
        "ssh-agent",
        "sh",
        "-c",
        "trap 'rm -f /runner/artifacts/29/ssh_key_data' EXIT && ssh-add /runner/artifacts/29/ssh_key_data && rm -f /runner/artifacts/29/ssh_key_data && ansible-playbook -t update_git,install_roles,install_collections -i /runner/inventory/hosts -e @/runner/env/extravars project_update.yml"
    ],
    "env": {
        "LAUNCHED_BY_RUNNER": "1",
        "ANSIBLE_UNSAFE_WRITES": "1",
        "AWX_ISOLATED_DATA_DIR": "/runner/artifacts/29",
        "ANSIBLE_FACT_CACHE_TIMEOUT": "0",
        "ANSIBLE_FORCE_COLOR": "True",
        "ANSIBLE_HOST_KEY_CHECKING": "False",
        "ANSIBLE_INVENTORY_UNPARSED_FAILED": "True",
        "ANSIBLE_PARAMIKO_RECORD_HOST_KEYS": "False",
        "HOME": "/var/lib/awx",
        "AWX_PRIVATE_DATA_DIR": "/tmp/awx_29_ywqhq5t9",
        "ANSIBLE_RETRY_FILES_ENABLED": "False",
        "ANSIBLE_ASK_PASS": "False",
        "ANSIBLE_BECOME_ASK_PASS": "False",
        "DISPLAY": "",
        "TMP": "/tmp",
        "PROJECT_UPDATE_ID": "29",
        "ANSIBLE_GALAXY_SERVER_SERVER0_URL": "https://galaxy.ansible.com/",
        "ANSIBLE_GALAXY_SERVER_LIST": "server0",
        "ANSIBLE_STDOUT_CALLBACK": "awx_display",
        "RUNNER_OMIT_EVENTS": "False",
        "RUNNER_ONLY_FAILED_EVENTS": "False"
    },
    "cwd": "/runner/project"
}
...
{"status": "successful", "runner_ident": "29"} 
```

So, the error is not reproducible alone by setting "AWX_TASK_ENV". The `callback` mount is still missing.

**Comment by anxstj** [comment]
I was thinking that the statement in https://github.com/ansible/awx/blob/devel/awx/main/tasks/jobs.py#L538 would solve (a part of) the mystery, but both systems have the same settings:

prod:
```
"IS_K8S": false,
```
```
$ psql awx -c 'select id,name,is_container_group from main_instancegroup;' | cat
 id |     name     | is_container_group 
----+--------------+--------------------
  2 | controlplane | f
  3 | default      | f
  4 | tower        | f
```

staging:
```
"IS_K8S": false
```
```
$ psql awx -c 'select id,name,is_container_group from main_instancegroup;' | cat
 id |     name     | is_container_group 
----+--------------+--------------------
  1 | controlplane | f
  2 | default      | f
```

**Comment by anxstj** [comment]
I tried to figure out why the ansible-runner version from AWX is older than the version from awx-ee. And it turned out that AWX installs directly the latest version from `devel`: See [requirements/requirements_git.txt](https://github.com/ansible/awx/blob/devel/requirements/requirements_git.txt#L3):
```
git+https://github.com/ansible/ansible-runner.git@devel#egg=ansible-runner
```
Unfortunately, the version number from devel is misleading, see: https://github.com/ansible/ansible-runner/issues/1041

@shanemcd This is not about a too old ansible-runner version. Something is mounting the callback plugin into the awx-ee container *and* the wrong HOME env variable was preventing it from being used. I would really like to understand what is causing the callback mount.

**Comment by anxstj** [comment]
I manually changed the ansible-runner version:
```
+++ b/requirements/requirements_git.txt
@@ -1,6 +1,6 @@
 git+https://github.com/ansible/system-certifi.git@devel#egg=certifi
 # Remove pbr from requirements.in when moving ansible-runner to requirements.in
-git+https://github.com/ansible/ansible-runner.git@devel#egg=ansible-runner
+git+https://github.com/ansible/ansible-runner.git@2.1.3#egg=ansible-runner
 # django-radius has an aggressive pin of future==0.16.0, see https://github.com/robgolding/django-radius/pull/25
 git+https://github.com/ansible/django-radius.git@develop#egg=django-radius
 git+https://github.com/PythonCharmers/python-future@master#egg=future
```

The awx image has then ansible-runner version 2.1.3:
```
(awx) bash-5.1$ pip freeze | grep ansible-runner
ansible-runner @ git+https://github.com/ansible/ansible-runner.git@e5a8b365e5c7ad2da3f3b1e0478aeed76bcab85f
(awx) bash-5.1$ ls -d /var/lib/awx/venv/awx/lib/python3.9/site-packages/ansible_runner*
ansible_runner/                 ansible_runner-2.1.3.dist-info/
```

The former image had:
```
(awx) bash-5.1$ pip freeze | grep ansible-runner
ansible-runner @ git+https://github.com/ansible/ansible-runner.git@49237b83365f2ccc3e0667dc0687d80d462d0788
(awx) bash-5.1$ ls -d /var/lib/awx/venv/awx/lib/python3.9/site-packages/ansible_runner*
ansible_runner/  ansible_runner-2.1.1.dev45.dist-info/
```

With the older version, the project checkout works, and the callback plugin is no longer mapped into the container.

Between these two versions `ansible_runner/callbacks/awx_display.py` was moved to `ansible_runner/display_callback/callback/awx_display.py`. And the function `callback_mount()` was added to `ansible_runner/utils/__init__.py`.

**To summarize:** In my special case with 'HOME' set as env variable, the two ansible-runner versions (devel (alias pre2.2.0) and 2.1.*) are incompatible. It's unfortunate that AWX installs another version than the awx-ee:latest image which is used by the control plane and which isn't (easily) changeable. But I guess, that's the risk if one is using the latest and greatest. :smiley: 

I will close the issue. The situation will solve itself with the next ansible-runner version (2.2).
